src/lib/token.c

```diff
@@ -228,9 +228,11 @@ CK_RV token_login(token *tok, twist pin, CK_USER_TYPE user) {
     twist pobjsalt = user == CKU_USER ? tok->userpobjauthkeysalt : tok->sopobjauthkeysalt;
     twist pobjauth = user == CKU_USER ? tok->userpobjauth : tok->sopobjauth;
 
-    dpobjauth = decrypt(pin, pobjsalt, pobjiters, pobjauth);
-    if (!dpobjauth) {
-        return CKR_PIN_INCORRECT;
+    if(pobjiters) {
```

Comment:
Wouldn't this be the encrypted null byte and then that get's passed to the tools (empty string) and it should work. All the functionality that is broken Is likely all in the tpm2_ptool.py script.

Checking pobjiters is not correct here. Also style on parens.

Reply:
@williamcroberts you are right. Checking pobjiters wasn't necessary.